packages/api/src/v2/resourcePlugin.ts

```diff
@@ -0,0 +1,235 @@
+  deploy?: (
+    ctx: Context,
+    provisionOutput: Readonly<ProvisionOutput>,
+    tokenProvider: AzureAccountProvider
+  ) => Promise<Result<{ output: Record<string, string> }, FxError>>;
+
+  /**
+   * Depends on the output of {@link package}. Uploads Teams package to AppStudio
+   * @param {Context} ctx - plugin's runtime context shared by all lifecycles.
+   * @param {AppStudioTokenProvider} tokenProvider - Token for AppStudio
+   * @param {Inputs} inputs - User answers to quesions defined in {@link getQuestionsForLifecycleTask}
+   * for {@link Stage.publish} along with some system inputs.
+   *
+   * @returns Void because side effect is expected.
+   */
+  publishApplication?: (
+    ctx: Context,
+    tokenProvider: AppStudioTokenProvider,
+    inputs: Inputs
+  ) => Promise<Result<Void, FxError>>;
+
+  /**
+   * provisionLocalResource is a special lifecycle, called when users press F5 in vscode.
+   * It works like provision, but only creates necessary cloud resources for local debugging like AAD and AppStudio App.
+   *
+   * @param {Context} ctx - plugin's runtime context shared by all lifecycles.
+   * @param {TokenProvider} tokenProvider - Tokens for Azure and AppStudio
+   *
+   * @returns the output values, project state, secrect values for the current environment. Toolkit will persist them
+   *          and pass them to {@link configureLocalResource}. The output will be persisted but not in the same file as provison's output.
+   */
+  provisionLocalResource?: (
+    ctx: Context,
+    tokenProvider: TokenProvider
+  ) => Promise<Result<LocalProvisionOutput, FxError>>;
+
+  /**
+   * configureLocalResource works like {@link configureResource} but only for local debugging resources.
+   * Plugins are expected to read the local provision output values of other plugins, and return a new copy of its own local provision output,
+   * possibly with added fields.
+   *
+   * @param {Context} ctx - plugin's runtime context shared by all lifecycles.
+   * @param {Readonly<LocalProvisionOutput>} localProvisionOutput - values generated by {@link provisionLocalResource}
+   * @param {Readonly<Record<PluginName, LocalProvisionOutput>>} provisionOutputOfOtherPlugins - values of other plugins generated by {@link provisionLocalResource}
+   * @param {TokenProvider} tokenProvider - Tokens for Azure and AppStudio
+   *
+   * @returns a new copy of provisionOutput possibly with added fields. The output will be persisted but not in the same file as provison's output.
+   */
+  configureLocalResource?: (
+    ctx: Context,
+    localProvisionOutput: Readonly<LocalProvisionOutput>,
+    localProvisionOutputOfOtherPlugins: Readonly<Record<PluginName, LocalProvisionOutput>>,
+    tokenProvider: TokenProvider
+  ) => Promise<Result<LocalProvisionOutput, FxError>>;
+
+  getQuestionsForLifecycleTask?: (
+    ctx: Context,
+    stage: Stage,
+    inputs: Inputs
+  ) => Promise<Result<QTreeNode | undefined, FxError>>;
+
+  getQuestionsForUserTask?: (
```

Comment:
getQuestionsForLifecycleTask: change to getQuestionForScaffold()
getQuestionsForUserTask: deleted
Just as we discussed, only scaffold questions can be extensible.